mlir/include/mlir/Dialect/Tosa/Transforms/TosaFoldCommon.h

```diff
@@ -40,10 +40,12 @@ DenseElementsAttr applyElementWise(
 /// Apply the given transformation function on the elements of the given
 /// tensors. If the input tensors do not match \p targetType, broadcasting is
 /// applied.
+template <class FirstValType, class SecondValType, class ResultType>
 DenseElementsAttr applyElementWise(
     const DenseElementsAttr &first, const DenseElementsAttr &second,
     TensorType targetType,
-    const std::function<APFloat(const APFloat &, const APFloat &)> &toApply);
+    const std::function<ResultType(const FirstValType &, const SecondValType &)>
```

Comment:
Are there binary ops where first and second operands have different types?

Reply:
Not really, should I change it to `ElementType`?

Reply:
yes, I think that would be clearer